pkg/apis/triggers/v1alpha1/scheduled_template_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2024 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+)
+
+// Check that ScheduledTemplate may be validated and defaulted.
+var _ apis.Validatable = (*ScheduledTemplate)(nil)
+var _ apis.Defaultable = (*ScheduledTemplate)(nil)
+
+// ScheduledTemplateSpec holds the desired state of ScheduledTemplate
+type ScheduledTemplateSpec struct {
+	// +listType=atomic
+	Params []ParamSpec `json:"params,omitempty"`
+	// +listType=atomic
+	ResourceTemplates []ScheduledResourceTemplate `json:"resourcetemplates,omitempty"`
+
+	CloudEventSink *apis.URL `json:"cloudEventSink,omitempty"`
+	Schedule       string    `json:"schedule,omitempty"`
```

Comment:
do you mind to add comments on these structs with explanation of its function and accepted format? It would make life of developer integrating this types way easier, thanks

Reply:
Sure, will do it.

Reply:
Done